### PR TITLE
Accessibility and labels in NPSP settings pages

### LINKED ISSUE #2003: Accessibility and labels in NPSP settings pages
**External Contribution Issue created by @shannonsans**

As part of the NPSP community sprint (Spring 16):

(1) Update the NPSP settings pages so that the help text is programmatically associated with the field for assistive technologies.
(2) Fix any hard-coded labels in the settings pages so they use Custom Labels instead.

### PR DESCRIPTION
**Pull request by mrbelvedere**
This change fixes most of the fields in the NPSP settings pages that have help text associated with them, by setting the html-aria-describedby passthrough attribute on the apex:select and apex:input* fields so that the help text is announced properly to screen readers and other assistive tech.

There's one remaining thing to be fixed, which I wanted to look at separately: this is the STG_DataBoundMultiSelect.component and all the places in which it's used in the settings. Because it hard-codes the ID inside the component, I didn't want to set the html-aria-describedby to an element not included within the component. The solution is to probably pass the help text message into the STG_DataBoundMultiSelect as an attribute so that it can be rendered with the rest of the component content, but this bears some additional research. So I'm submitting this part for now and will look at the remaining issue in a separate request.

# Contribution Info

This external contribution was submitted by @shannonsans and Fixes #2003

# Critical Changes

Fields in the NPSP Settings pages now have help text blurbs programmatically associated with their respective fields, so that assistive technologies such as screen readers can contextually announce the help text. This change only impacts admin users who need to use a screen reader or similar assistive technology to interact with the page.

# Changes

This change is automatically enabled for users who have access to the STG_SettingsManager page.

# Issues

Fixes #2003

### CODE REVIEW

File: src/pages/STG_PanelContactRoles.page [hunk 1]
```diff
@@ -12,17 +12,21 @@
                 <apex:outputLabel value="{!$ObjectType.npe01__Contacts_And_Orgs_Settings__c.Fields.npe01__Opportunity_Contact_Role_Default_role__c.Label}" for="slOCRDR" styleClass="slds-form-element__label" />
                 <div class="slds-form-element__control">
                     <apex:outputField value="{!stgService.stgCon.npe01__Opportunity_Contact_Role_Default_role__c}" rendered="{!isReadOnlyMode}" styleClass="slds-form-element__static" />
-                    <apex:selectList value="{!stgService.stgCon.npe01__Opportunity_Contact_Role_Default_role__c}" multiselect="false" size="1" rendered="{!isEditMode}" id="slOCRDR" styleClass="slds-select" >
+                    <apex:selectList value="{!stgService.stgCon.npe01__Opportunity_Contact_Role_Default_role__c}" multiselect="false" size="1" rendered="{!isEditMode}" id="slOCRDR" html-aria-describedby="{!$Component.slOCRDRHelp}" styleClass="slds-select" >
```
Comment (davidhabib) [inline src/pages/STG_PanelContactRoles.page]: Should this be use $Label.slOCRDRHelp, not $Component.slOCRDRHelp?

File: src/pages/STG_PanelContactRoles.page [hunk 2]
```diff
@@ -31,25 +35,31 @@
...
+                        <apex:outputText styleClass="slds-form-element__help" value="{!$Label.stgHelpSoftCreditRoles}" />
+                    </apex:outputPanel>
                 </div>
             </div>
             <div class="slds-form-element">
                 <apex:outputLabel value="{!$ObjectType.npo02__Households_Settings__c.Fields.Matched_Donor_Role__c.Label}" for="slMDR" styleClass="slds-form-element__label" />
                 <div class="slds-form-element__control">
                     <apex:outputField value="{!stgService.stgHH.Matched_Donor_Role__c}" rendered="{!isReadOnlyMode}" styleClass="slds-form-element__static" />
-                    <apex:selectList value="{!stgService.stgHH.Matched_Donor_Role__c}" multiselect="false" size="1" rendered="{!isEditMode}" id="slMDR" styleClass="slds-select" >
+                    <apex:selectList value="{!stgService.stgHH.Matched_Donor_Role__c}" multiselect="false" size="1" rendered="{!isEditMode}" id="slMDR" html-aria-describedby="{!$Component.slMDRHelp}" styleClass="slds-select" >
```
Comment (davidhabib) [inline src/pages/STG_PanelContactRoles.page]: Shouldn't this be using $Label.slMDRHelp, not $Component.slMDRHelp?

**Comment by davidhabib** [comment]
@mpusto for sprint planning, I think this should have 3 story points for a dev on the team to get into their branch without some of the invalid metadata changes (due to the contrib tool, not the author!), and test it out.

@shannonsans ignore my earlier comments about html-aria-describedby!  I misunderstood its purpose.

**Comment by njjc** [comment]
Moved over to #2063

**Pull request closed without merging**